javascript-client/src/client.ts

```diff
@@ -46,13 +46,44 @@ export interface ProgressEvent {
 
 type ProgressCallback = (progress: ProgressEvent) => void;
 
+export interface S3FFClientOptions {
+  readonly onProgress?: ProgressCallback;
+  readonly apiInstance?: AxiosInstance;
+  readonly apiConfig?: AxiosRequestConfig
+}
+
 export default class S3FFClient {
+  protected readonly api: AxiosInstance;
+
+  protected readonly onProgress: ProgressCallback;
+
+  /**
+   * Create an S3FFClient instance.
+   *
+   * @param baseUrl - The absolute URL to the Django server.
+   * @param [options] {S3FFClientOptions} - Optional arguments.
+   * @param [options.onProgress] - A callback for upload progress.
+   * @param [options.apiInstance] - An axios instance to use for Django API requests.
```

Comment:
As written, this doesn't use the passed axios instance. It uses a copy of its configuration, with a modified baseUrl. Which is kind of why I preferred just passing in a configuration object rather than an instance.

Reply:
I don't have strong opinions here. Any thoughts, @danlamanna ?

Reply:
@zachmullen Do you think we should cut this then?

I kinda need someone to push for something. I've got some decision fatigue myself. :smiley:

Reply:
I don't have more to say really. I think we're not going to know what we want the API to look like until we start using it in the wild. Pick something and roll with it. Attempting to achieve the perfect API this early is silly.